File: reservation-service/src/main/java/org/sdia/reservationservice/entities/Reservation.java
package org.sdia.reservationservice.entities;

import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.persistence.*;
import lombok.*;
import org.sdia.reservationservice.model.Ressource;

import java.util.Date;

@Entity
@Getter @Setter @AllArgsConstructor @NoArgsConstructor @Builder
public class Reservation {
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String nom;
    private String contexte;
    private Date date;
    private String dure;
    private Long ressourceId;
    @Transient
    private Ressource ressource;
    @ManyToOne
    @JsonProperty(access = JsonProperty.Access.WRITE_ONLY)
    private Personne personne;
}

File: app-ng/src/app/Reservation.ts
import {Ressource} from "./Ressource";
import {Personne} from "./Personne";

export interface Reservation{

    id : number;
    nom : string;
    contexte:string;
    dure:string;
    date:Date;
    ressourceId:number;
    ressource:Ressource;
    personne:Personne;
}


